plugins/tech-insights-backend-module-jsonfc/src/service/JsonRulesEngineFactChecker.ts

```diff
@@ -272,7 +275,7 @@ export class JsonRulesEngineFactChecker
   ) {
     const returnable = {
       id: techInsightCheck.id,
-      type: techInsightCheck.type,
+      type: BOOLEAN_CHECK_RESPONSE_TYPE,
```

Comment:
By doing this we lose the information that the check is embedded in the returned data type. The ability to render informational information about the check itself is not implemented on the current frontend result displaying components but would be beneficial to display for some cases. 

Apart from making the typings more generic, what other reason would there be for this change?

Reply:
I think that using the same type for check results and for JSON rules engine checks makes the frontend plugin (which should be generic) too much tightly coupled to the backend JSON rules engine module (which we should consider optional; it's a module after all, and a user can use different implementation). I think that if there were another frontend module explicitly related to the JSON rules engine, that would make sense to use the same type. 

The pragmatic benefit we get making it more generic is that a user can use another fact checker returning 'boolean' results and doesn't need to do any changes to the frontend.

As an alternative, if you don't want to change the result type, I think we can remove the default renderer from the frontend with the type `json-rules-engine` and add a paragraph in the docs on how to add a generic `booleanCheckResultRenderer` for the `json-rules-engine` type if a user chooses this implementation. That would make the prior mentioned connection between the frontend and the backend module less explicit.

Reply:
The coupling is for the return type of the returned check results and not the module itself in this case. The purpose of the type property is to define what data the response contains and having it explicitly stated, rather than implicitly guess based on properties present on the object is a better approach IMO.

I'm understanding that you want to enable the possibility to use the same frontend components with a different implementation of a check engine. I think the approach to implement renderer injection during the API implementation initialization is the correct one and changes on the FE module are fine. 

I don't think I'm comfortable making a breaking change on the produced API signature due to this change because it hides valuable information from the payload that is in use.

Reply:
OK, I reverted the changes to the backend and the JSON rules engine module. All the changes are now only on the frontend to allow for registering different renderers for different types.